core/src/session.rs

```diff
@@ -284,7 +301,11 @@ impl Session {
                 }
                 .as_secs() as i64;
 
-                self.0.data.write().time_delta = server_timestamp - timestamp;
+                {
+                    let mut data = self.0.data.write();
+                    data.time_delta = server_timestamp - timestamp;
```

Comment:
Might want to do a saturating sub here just in case.

Reply:
Sounds good. I also changed the `timestamp` to be `Duration::ZERO`, if `SystemTime::now()` is before `UNIX_EPOCH`.